src/main/java/fi/dy/masa/servux/event/PlayerHandler.java

```diff
@@ -0,0 +1,101 @@
+package fi.dy.masa.servux.event;
+
+import javax.annotation.Nullable;
+import java.net.SocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import com.mojang.authlib.GameProfile;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import fi.dy.masa.servux.interfaces.IPlayerListener;
+
+public class PlayerHandler implements IPlayerManager
+{
+    private static final PlayerHandler INSTANCE = new PlayerHandler();
+    private final List<IPlayerListener> handlers = new ArrayList<>();
+    public static IPlayerManager getInstance() { return INSTANCE; }
+
+    @Override
+    public void registerPlayerHandler(IPlayerListener handler) {
+        if (!this.handlers.contains(handler))
+        {
+            this.handlers.add(handler);
+        }
+    }
+
+    @Override
+    public void unregisterPlayerHandler(IPlayerListener handler)
+    {
+        this.handlers.remove(handler);
+    }
+
+    /**
+     * NOT PUBLIC API - DO NOT CALL
+     */
+    public void onClientConnect(SocketAddress addr, GameProfile profile, @Nullable Text result)
```

Comment:
Consider `org.jetbrains.annotations.ApiStatus.Internal`

Reply:
Again, I am not here to change masa's coding style.